src/Widgets/Views/MainView.vala

```diff
@@ -90,11 +94,11 @@ public class Bluetooth.Widgets.MainView : Gtk.Box {
         });
 
         object_manager.device_removed.connect ((device) => {
-            devices_box.get_children ().foreach ((child) => {
-                if (child is Bluetooth.Widgets.Device) {
-                    ((Bluetooth.Widgets.Device) child).destroy ();
-                }
-            });
+            if (devices.has_key (device.modalias)) {
```

Comment:
IIRC, doing that with a HashMap returned a warning if the key wasn't there.... But i'll give it a try! I forgot `hash_map[key]` was a thing :)